beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -1298,6 +1299,33 @@ impl<T: BeaconChainTypes> Worker<T> {
         metrics::inc_counter(&metrics::BEACON_PROCESSOR_SYNC_CONTRIBUTION_IMPORTED_TOTAL);
     }
 
+    pub fn process_gossip_finality_update(
+        self,
+        message_id: MessageId,
+        peer_id: PeerId,
+        light_client_finality_update: LightClientFinalityUpdate<T::EthSpec>,
+        seen_timestamp: Duration,
+    ) {
+        match self
+            .chain
+            .verify_finality_update_for_gossip(light_client_finality_update, seen_timestamp)
+        {
+            Ok(verified_light_client_finality_update) => {
+                self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Accept);
+            }
+            Err(e) => {
+                debug!(
+                    self.log,
+                    "Invalid light client finality update message";
+                    "peer" => %peer_id,
+                    "error" => ?e
+                );
+                self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Ignore);
```

Comment:
I feel that we would want to downscore peers on some error variants here.

- **TooEarly**: From what I understand, this particular error variant is to ensure that we complete the important duties pertaining to block verification and setting head before processing lower priority light client tasks. Getting way too many `TooEarly` errors could indicate that the peer's clock is not synced. It doesn't seem like we do any expensive processing if we encounter this error so doesn't seem like a DOS risk, but we would still like to discourage such behaviour by propagating a `HighTolerance` error.
- **InvalidLightClientFinalityUpdate**: This should result in a reject at the gossip layer and a LowTolerance lighthouse peer score action.

You can check the `handle_sync_committee_message_failure` function in this file to check how we handle the error variants.

Reply:
Thank you for your feedback! The spec shows: [IGNORE] The received finality_update matches the locally computed one exactly (as defined in create_light_client_finality_update) . The InvalidLightClientFinalityUpdate error is supposed to implement that check. Are you suggesting it should be [REJECT]?

Reply:
Ahh apologies, I checked the light client validations instead of the server validations in the spec. The condition seems correct in that case. Nevertheless, I would prefer to add some lighthouse specific peer scoring depending on the errors as well.
Something like
```rust
match err {
        Error::InvalidLightClientFinalityUpdate | Error::TooEarly => self.gossip_penalize_peer(
            peer_id,
            PeerAction::HighToleranceError,
            "light_client_gossip_error",
        ),
        Error::BeaconChainError(_)
        | Error::FinalityUpdateAlreadySeen
        | Error::SigSlotStartIsNone => {}
    }
```